### ISSUE TITLE
Packages with security vulnerabilities in UI

### ISSUE THREAD

**Issue by mayzhang2000** [argoproj/argo-cd#6917]
If you are trying to resolve an environment-specific issue or have a one-off question about the edge case that does not require a feature then please consider asking a question in argocd slack [channel](https://argoproj.github.io/community/join-slack).

Checklist:

* [ ] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [ ] I've included steps to reproduce the bug.
* [ ] I've pasted the output of `argocd version`.

**Describe the bug**
The Argo CD UI uses outdated and insecure dependencies that have high and critical vulnerabilities. 
The full list of vulnerable packages can be seen by invoking the ​npm audit​ tool​ within the respective u​ i​ directory of the Argo CD project. 
 
~/argo-cd/ui $ npm audit --level=moderate
40904 vulnerabilities found - Packages audited: 1644
Severity: 40878 Low | 18 Moderate | 8 High
Done in 6.55s.

**To Reproduce**

A list of the steps required to reproduce the issue. Best of all, give us the URL to a repository that exhibits this issue.

**Expected behavior**

**Screenshots**

**Version**

```shell
Paste the output from `argocd version` here.
```

**Logs**

```
Paste any relevant application logs here.
```

**Comment by harikrongali** [comment]
current vulnerabilities
 ```
9 vulnerabilities found - Packages audited: 1468
Severity: 1 Low | 2 Moderate | 6 High
```

**Comment by alexmt** [comment]
Just to clarify. Here is output of `yarn audit`. As you might see all vulnerabilities are dev tools dependencies and not even used by application. Only one vulnerability `normalize-url` URL is used by UI but totally not practical. Only admins can register Git repos, so if someone get access to register repo then slow UI is not so important issue.

<details>
<summary>yarn audit</summary>
<p>



```
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ low           │ Prototype Pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ yargs-parser                                                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=13.1.2 <14.0.0 || >=15.0.1 <16.0.0 || >=18.1.2             │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ ts-jest                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ ts-jest > yargs-parser                                       │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://www.npmjs.com/advisories/1500                        │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ moderate      │ Regular expression denial of service                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ glob-parent                                                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=5.1.2                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ webpack                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ webpack > watchpack > watchpack-chokidar2 > chokidar >       │
│               │ glob-parent                                                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://www.npmjs.com/advisories/1751                        │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ high          │ Regular Expression Denial of Service                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ normalize-url                                                │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=4.5.1 <5.0.0 || >=5.3.1 <6.0.0 || >=6.0.1                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ git-url-parse                                                │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ git-url-parse > git-up > parse-url > normalize-url           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://www.npmjs.com/advisories/1755                        │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ high          │ Regular Expression Denial of Service                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ diff                                                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >= 3.5.0                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ unidiff                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ unidiff > diff                                               │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://www.npmjs.com/advisories/1631                        │
└───────────────┴──────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ moderate      │ Regular expression denial of service                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ glob-parent                                                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=5.1.2                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ webpack-dev-server                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ webpack-dev-server > chokidar > glob-parent                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://www.npmjs.com/advisories/1751                        │
└───────────────┴──────────────────────────────────────────────────────────────┘
```

</p>
</details>  

We might try to fix these vulnerabilities but will get new ones tomorrow. This article describes well problem with `yarn/npm audit` :  https://overreacted.io/npm-audit-broken-by-design/

**Comment by alexmt** [comment]
Discussed with the team and agreed to introduce one more release procedure: before every release team should triage all critical vulnerabilities reported by the `yarn audit` and ensure that there are no exploitable ones.
